[docs-infra] Add small design polish

danilo-leal opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

Some simple stuff on this PR — here's a run-down:

- Actually, maybe the most significant change of this PR is removing the logo and extra links from the docs footer. I had added that many months ago, but ultimately, I feel like they just clutter the docs without a great reason. Way cleaner without them!
- Swapping some semi-bold font-weight instances to medium to lighten things up a bit
- A bunch of color improvements, mostly on the icon button (improving shadows, border, and hover state styles)

Preview: https://deploy-preview-42455--material-ui.netlify.app/material-ui/migration/migration-v5/

zanivan reviewed (approved):
LGTM! It looks way cleaner!

colmtuite commented:
I give a thumbs up to making text lighter for sure.

Do people click those footer links much? Are we tracking those clicks? Seems like a potential regression to me to remove those social links, and probably the store link too. I found the store link in the left sidebar, but can't find a way to get to Twitter or Discord now.

Don't the "edit this page" and "was this page helpful" buttons make more sense above the next/previous links?

zanivan commented:
> Do people click those footer links much? Are we tracking those clicks?

I don't think we were tracking these--at least I couldn't find anything in GA or in the buttons itself. 

About discord--we aren't giving it too much attention, so I don't know how important this link here is. IMO twitter is likely the only possible regression here.

> Don't the "edit this page" and "was this page helpful" buttons make more sense above the next/previous links?

+1 on this. I think this makes a lot of sense, actually. Should we try a variation like Geist?

![image](https://github.com/mui/material-ui/assets/37222944/28250ce2-6d32-46d2-9004-4230c7f03311)

danilo-leal commented:
Unsure how much we should be worried about Twitter and Discord traffic coming from the docs footer. At least Twitter, which I follow more closely, didn't grow too significantly after I introduced these links on the footer (it wasn't a long while ago). But having the Store link was maybe indeed good, and an individual event tracking traffic coming from this link would've been ideal, yeah. In any case, not super confident that Store link drove a lot of access/sales... but If y'all want to err on the side of caution, I'm okay with putting it back.

The "edit this page" and "was this helpful" stuff was below the page buttons already, changed here as it felt better for whatever reason 🤷 maybe yeah, leaving it as is make sense, ha. I wouldn't change the layout too drastically, closer to Gesit or other design, here yet, though.

colmtuite commented:
We will need Discord soon for Base UI. Well, I mean, hopefully we will need it. If the product becomes popular, we will need some support channel, and Discord is the best option. So I'd keep links to that visible.

> The "edit this page" and "was this helpful" stuff was below the page buttons already

I'm not understanding this part. On the live docs, the "edit this page" and "was this helpful" stuff is currently above the page links, not below. Or are you referring to a long time ago?

danilo-leal commented:
Okay — rolled back a few things on the footer: icons and links are back.

colmtuite commented:
We're so back
